### PR TITLE
Migrate bcrypt APIs

### PR DESCRIPTION
**Pull request by skdltmxn**

### CODE REVIEW

File: src/shared/bcrypt.rs [hunk 1]
```diff
@@ -84,6 +84,51 @@ STRUCT!{struct BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO {
     dwFlags: ULONG,
 }}
 pub type PBCRYPT_AUTHENTICATED_CIPHER_MODE_INFO = *mut BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO;
+// BCrypt String Properties
+pub const BCRYPT_OPAQUE_KEY_BLOB: &'static str = "OpaqueKeyBlob";
+pub const BCRYPT_KEY_DATA_BLOB: &'static str = "KeyDataBlob";
+pub const BCRYPT_AES_WRAP_KEY_BLOB: &'static str = "Rfc3565KeyWrapBlob";
+
+// BCryptGetProperty strings
```
Comment (retep998) [inline src/shared/bcrypt.rs]: I'd prefer not to have comments and blank lines littered about.

**Comment by skdltmxn** [comment]
Removed all evil blanks :)

File: src/shared/bcrypt.rs [hunk 2]
```diff
@@ -254,18 +502,278 @@ STRUCT!{struct BCRYPT_ALGORITHM_IDENTIFIER {
...
 STRUCT!{struct BCRYPT_INTERFACE_VERSION {
     MajorVersion: USHORT,
     MinorVersion: USHORT,
 }}
 pub type PBCRYPT_INTERFACE_VERSION = *mut BCRYPT_INTERFACE_VERSION;
+#[inline]
+pub fn BCRYPT_IS_INTERFACE_VERSION_COMPATIBLE(
+    loader: BCRYPT_INTERFACE_VERSION,
+    provider: BCRYPT_INTERFACE_VERSION
+    ) -> bool {
```
Comment (retep998) [inline src/shared/bcrypt.rs]: De-indent this line.

Comment (retep998) [inline src/shared/bcrypt.rs]: Also don't forget the trailing comma above.

**Comment by skdltmxn** [comment]
Fixed :)

**Comment by retep998** [comment]
Could you update your three PRs now that I fixed the issue causing the CI failures?

**Pull request merged**